src/NLog/Common/ConversionHelpers.cs

```diff
@@ -44,24 +44,39 @@ public static class ConversionHelpers
         /// <summary>
         /// Converts input string value into <see cref="System.Enum"/>. Parsing is case-insensitive.
         /// </summary>
-        /// <param name="inputValue">Input value</param>
-        /// <param name="resultValue">Output value</param>
+        /// <param name="value">Input value</param>
+        /// <param name="result">Output value</param>
         /// <param name="defaultValue">Default value</param>
-        /// <returns>Returns failure if the input value could not be parsed</returns>
-        public static bool TryParseEnum<TEnum>(string inputValue, out TEnum resultValue, TEnum defaultValue = default(TEnum)) where TEnum : struct
+        /// <returns>Returns <paramref name="defaultValue"/> if the input value could not be parsed</returns>
+        public static bool TryParseEnum<TEnum>(string value, out TEnum result, TEnum defaultValue = default(TEnum)) where TEnum : struct
         {
-            if (string.IsNullOrEmpty(inputValue))
+            if (!TryParseEnum(value, true, out result))
```

Comment:
Are you not changing behavior of public-method when removing the string.IsNullOrEmpty-check ?

Reply:
yes, it's a bug IMO

Reply:
Can see you have adjusted in both DatabaseParameterInfo.cs  and MailTarget.cs to cater for the new behavior. What about Mailkit ?

Reply:
But must admit I also prefer that it returns false when empty-string. But one should just have special logging when parsing fails, and fallback is performed.